### PR TITLE
add CentOS support("PR is released under MIT License")

### PR DESCRIPTION
**Pull request by HeyManLeader**
There is a delicate difference between Ubuntu and CentOS. Add CentOS7 support in openvstorage-health-check.

### CODE REVIEW

**Comment by wimpers** [comment]
Hi, 

thank you for your contribution. We can't accept the PR alas as you didn't sign our Contributor License Agreement (yet).

Please see https://www.openvstorage.com/contribute/ .

File: ovs/extensions/healthcheck/utils/extension.py [hunk 1]
```diff
@@ -184,8 +184,19 @@ def check_etcd(self):
...
+            release_version_return = self.execute_bash_command('cat /etc/os-release')
+            # All OS distribution classes used in below code should share the exact same interface!
+            if 'CentOS Linux' in release_version_return[0]:
+                result = self.execute_bash_command("rpm -qa | grep etcd")
+            elif 'Ubuntu' in release_version_return[0]:
+                result = self.execute_bash_command("dpkg -l | grep etcd")
+            else:
+                raise RuntimeError('The OS is not supported')
+        except Exception as ex:
+            raise RuntimeError('Error checking etcd: {0}'.format(ex))
```
Comment (khenderick) [inline ovs/extensions/healthcheck/utils/extension.py]: Better not catch the exception, but let it fall through so we don't lose stacktrace information.

File: ovs/extensions/healthcheck/openvstorage/openvstoragecluster_health_check.py [hunk 2]
```diff
@@ -846,12 +859,13 @@ def check_model_consistency(self):
...
+                # this can happen , there is difference between Ubuntu and CentOS
+                # get the partition info from the last item
                 if len(cluster_status) <= 3:
-                    partition_status = cluster_status[2]
+                    partition_status = cluster_status[-1]
                 else:
-                    partition_status = cluster_status[3]
-
+                    partition_status = cluster_status[-1]
+
```
Comment (khenderick) [inline ovs/extensions/healthcheck/openvstorage/openvstoragecluster_health_check.py]: Unnecessary spaces

File: ovs/extensions/healthcheck/openvstorage/openvstoragecluster_health_check.py [hunk 3]
```diff
@@ -368,14 +368,27 @@ def check_ovs_packages(self):
...
-                self.LOGGER.logger(
-                    "Package '%s' is present, with version '%s'" % (package, result[2].split(':')[1].strip()),
-                    self.module, 1, 'package_{0}'.format(package))
-            else:
-                self.LOGGER.logger("Package '{0}' is NOT present ...".format(package), self.module, 5,
-                                   'package_{0}'.format(package))
+            try:
+                release_version_return = self.utility.execute_bash_command('cat /etc/os-release')
+                if 'CentOS Linux' in release_version_return[0]:
+                    result = self.utility.execute_bash_command("rpm -q %s" % package)
```
Comment (khenderick) [inline ovs/extensions/healthcheck/openvstorage/openvstoragecluster_health_check.py]: Please use new-style string formatting (same for the below statements)

**Comment by khenderick** [comment]
Thanks for your contributions. I already added some remarks.

However, before we can accept this contribution, we need you to sign a Contributor License Agreement. Please find the information about this at http://openvstorage.org/contribute/

**Comment by wimpers** [comment]
In case you don't want or can't sign the CLA, please explicitly indicate you want to release the PR under the MIT license by adding here as comment:

"Pull Request released under the MIT license."

File: ovs/extensions/healthcheck/openvstorage/openvstoragecluster_health_check.py [hunk 4]
```diff
@@ -366,29 +366,23 @@ def check_ovs_packages(self):
...
                 else:
-                    self.LOGGER.logger('The OS is not supported', self.module, 0, 'check_os_release')
-                    raise RuntimeError('The OS is not supported')
-            except Exception as ex:
-                self.LOGGER.logger('Error checking os-release: {0}'.format(ex), self.module, 0, 'check_os_release')
-                raise
-
+                    self.LOGGER.logger("Package '{0}' is NOT present ...".format(package), self.module, 5, 'package_{0}'.format(package))
+            elif "Ubuntu" in release_version_return[0]:
+                result = self.utility.execute_bash_command("apt-cache policy %s" % package)
```
Comment (kinvaris) [inline ovs/extensions/healthcheck/openvstorage/openvstoragecluster_health_check.py]: Please use: `result = self.utility.execute_bash_command("apt-cache policy {0}".format(package))`

**Pull request closed without merging**